### PR TITLE
fix: unread event badge [AR-2791]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2791" title="AR-2791" target="_blank">BugAR-2791</a>  New Activity section is not showing
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
- fixed new activity section
- fixed performance for mapping conversations by executing them on the io thread
- removed unused fields 

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1212/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1212/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5144494)(5144494)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5144494)(5144494)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1212/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5144833)(5144833)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5144833)(5144833)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 1]
```diff
@@ -105,45 +105,91 @@ class ConversationListViewModel @Inject constructor(
...
     }
 
     // Mateusz : First iteration, just filter stuff
     // next iteration : SQL- query ?
-    private fun searchConversation(conversationDetails: List<ConversationDetails>, searchQuery: String): List<ConversationDetails> {
+    private fun searchConversation(conversationDetails: List<ConversationItem>, searchQuery: String): List<ConversationItem> {
         val matchingConversations = conversationDetails.filter { details ->
-            details.conversation.name?.contains(searchQuery, true) ?: false
+            when (details) {
+                is ConversationItem.ConnectionConversation -> details.conversationInfo.name.contains(searchQuery, true)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:140]: what about searching for the user handle?

**Review by saleniuk — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:140]: Created thicket for that

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1212/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5145631)(5145631)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5145631)(5145631)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:140]: **thick**et because it's so huge topic? 😄

**Pull request merged**